services/QuillLMS/app/controllers/auth/google_controller.rb

```diff
@@ -1,28 +1,48 @@
 class Auth::GoogleController < ApplicationController
-  before_action :set_profile,                     only: :google
-  before_action :set_user,                        only: :google
-  before_action :save_teacher_from_google_signup, only: :google
-  before_action :save_student_from_google_signup, only: :google
-  before_action :follow_google_redirect,          only: :google
-
-  def google
-    if @user.teacher?
-      GoogleStudentImporterWorker.perform_async(@user.id, 'Auth::GoogleController')
-    end
+  before_action :set_profile, only: [:authorization_and_authentication, :authentication]
+  before_action :verify_credentials, only: :authentication
+  before_action :set_user,
+    :save_teacher_from_google_signup,
+    :save_student_from_google_signup,
+    :follow_google_redirect,
+    only: [:authorization_and_authentication, :authentication]
 
-    if @user.student?
-      GoogleStudentClassroomWorker.perform_async(@user.id)
-    end
+  # The reason for the separate 'authentication' and`authorization_and_authentication` methods lies in the
+  # before_action hook :verify_credentials.  This is only called with the authentication flow and if it fails
+  # will redirect to `google_oauth2` for reauthorization.
 
+  def authorization_and_authentication
```

Comment:
Outside the scope of this PR, but separating the "create account" and the "login" even more, might be helpful in future refactors / cleaning up the code, e.g. there's a lot of code in these `before_action`s that branches on `@user.new_record?`

Reply:
I agree.  I think all the branching begs for this to be broken out into two parts.  However, I was reluctant to do that given the lack of test coverage.  I think once we get some integration testing in place, I would feel more comfortable cleaning this up further.